### PR TITLE
[MIGRATION] Add note about authentication errors and server errors

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [ ] Added tests for code changes _or_ test/build only changes
- [ ] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description
There have been some questions regarding migration lately. This PR tries to address them.

## Approach

I clarified some of the questionable topics in this change:
* where users should check the server errors and the authentication errors

I also fixed a typo from python-java-sdk to cloudant-python-sdk

My IDE lately breaks the lines automatically, I have not figured out how I can turn it off yet, but I think this is a great feature, and it helps the reviewers to read the long lines more effectively, so I kept these modifications.

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- N/A

## Monitoring and Logging

- "No change"

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -20,7 +20,13 @@ content like using `raw_result=True` in `python-cloudant`.
 1. Replay adapters are replaced by the [automatic retries](https://github.
    com/IBM/ibm-cloud-sdk-common/#automatic-retries) feature for failed requests.
 1. Error handling is not transferable from `python-cloudant` to `cloudant-python-sdk`. For more information go to the [Error handling section](https://cloud.ibm.com/apidocs/cloudant?code=python#error-handling) in our API docs.
-1. Custom HTTP client configurations in `python-cloudant` are not transferable to `python-java-sdk`. For more information go to the [Configuring the HTTP client section(https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
+1. Custom HTTP client configurations in `python-cloudant` are not transferable to 
+   `cloudant-python-sdk`. For more information go to the [Configuring the HTTP client section
+   (https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
```
Comment (vmatyusGitHub) [inline MIGRATION.md:25]:
```suggestion
   (https://github.com/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
```

Comment (mojito317) [reply MIGRATION.md]: Done in d8a1756

File: MIGRATION.md [hunk 2]
```diff
@@ -20,7 +20,14 @@ content like using `raw_result=True` in `python-cloudant`.
 1. Replay adapters are replaced by the [automatic retries](https://github.
    com/IBM/ibm-cloud-sdk-common/#automatic-retries) feature for failed requests.
 1. Error handling is not transferable from `python-cloudant` to `cloudant-python-sdk`. For more information go to the [Error handling section](https://cloud.ibm.com/apidocs/cloudant?code=python#error-handling) in our API docs.
-1. Custom HTTP client configurations in `python-cloudant` are not transferable to `python-java-sdk`. For more information go to the [Configuring the HTTP client section(https://githubcom/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud SDK Common README.
+1. Custom HTTP client configurations in `python-cloudant` are not transferable to 
+   `cloudant-python-sdk`. For more information go to the [Configuring the HTTP client section
+   (https://github.com/IBM/ibm-cloud-sdk-common/#configuring-the-http-client) in the IBM Cloud 
+   SDK Common README.
+1. Authentication errors turn out at the time of instantiation of a client, while errors
```
Comment (vmatyusGitHub) [inline MIGRATION.md:27]: Would you plan to update this chapter too, like you did with java and nodejs?

Comment (mojito317) [reply MIGRATION.md:27]: yes, I just need to figure out what is the authentication error message first

Comment (mojito317) [reply MIGRATION.md]: Done in https://github.com/cloudant/python-cloudant/pull/512/commits/15a9b609719fb907efa72214a2ab9d050e3c37f5

**Review by vmatyusGitHub — APPROVED** [review]
LGTM :+1:

**Review by emlaver — APPROVED** [review]

**Comment by mojito317** [comment]
Thanks for the reviews @emlaver, @vmatyusGitHub! I am merging this!

**Pull request merged**